it/pages/contactPreference/LetterPreferencePageSpec.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.contactPreference
+
+import common.SessionKeys
+import forms.YesNoForm
+import helpers.SessionCookieCrumbler
+import models.contactPreferences.ContactPreference.digital
+import models.{No, Yes, YesNo}
+import pages.BasePageISpec
+import play.api.http.Status
+import play.api.libs.ws.WSResponse
+import stubs.VatSubscriptionStub
+
+class LetterPreferencePageSpec extends BasePageISpec {
+
+  val path = "/contact-preference-letter"
+
+  "Calling the LetterPreference .show method" when {
+
+    def show: WSResponse = get(path, formatCurrentContactPref(Some(digital)))
+
+    "the user is authenticated" should {
+
+      "display the page correctly" in {
+
+        given.user.isAuthenticated
+
+        And("a successful response for an individual is stubbed")
+        VatSubscriptionStub.stubCustomerInfo
+
+        val result = show
+
+        result should have(
+          httpStatus(Status.OK),
+          pageTitle(generateDocumentTitle("letterPreference.title"))
+        )
+      }
+    }
+
+    "a user is an agent" should {
+
+      "redirect to Agent overview" in {
```

Comment:
Inaccurate description

Reply:
This one's correct isn't it?

Reply:
It's going to the client VAT number page not agent overview

Reply:
Yeah damn sorry, you're right, I've completely misread that.